kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/util/KubernetesUtils.java

```diff
@@ -161,6 +165,44 @@ public static PropertySource secretAsPropertySource(Secret secret) {
         return excludesFilter;
     }
 
+    /**
+     * @param client       the {@link KubernetesClient}
+     * @param podLabelKeys the list of labels inside a pod
+     * @param namespace    in the configuration
+     * @return the filtered labels of the current pod
+     */
+    public static Flowable<Map<String, String>> computePodLabels(KubernetesClient client, List<String> podLabelKeys, String namespace) {
+        // determine if we are running inside a pod. This environment variable is always been set.
+        String host = System.getenv(ENV_KUBERNETES_SERVICE_HOST);
+        if (host == null) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Not running on k8s");
+            }
+            return Flowable.just(new HashMap<>());
+        }
+
+        return Flowable.fromPublisher(client.getPod(namespace, System.getenv(HOSTNAME_ENV_VARIABLE))).map(
+                pod -> {
+                    LOG.info("HELLO FROM THE FCKING SINGLE");
```

Comment:
This could be removed 😄

Reply:
Oh yeah, my first steps with rxJava :sweat_smile: